doc/glossary.rst

```diff
@@ -1666,10 +1666,24 @@ functions or non-estimator constructors.
         in a subsequent call to :term:`fit`.
 
         Note that this is only applicable for some models and some
-        parameters, and even some orders of parameter values. For example,
-        ``warm_start`` may be used when building random forests to add more
-        trees to the forest (increasing ``n_estimators``) but not to reduce
-        their number.
+        parameters, and even some orders of parameter values. In general, there
+        is an interaction between ``warm_start`` and the parameter controlling
+        the number of iterations of the estimator.
+
+        With ensemble estimators (i.e. estimators imported from :mod:`ensemble`),
+        ``warm_start`` will interact with ``n_estimators`` or ``max_iter``.
+        For these models, the number of iterations, reported via
+        ``len(estiamtors_)`` or ``n_iter_``, corresponds the total number of
+        iterations/estimators learn since the intiailization of the model.
+        Thus, if a model was already initialized with `N`` estimators, and `fit`
+        is called with ``n_estimators`` or ``max_iter`` set to `M`, the model
+        will train `M - N` new estimators.
+
+        Other models have a different behavior. They all expose a ``max_iter``
```

Comment:
Does "Other models" mean "models that do not warm start"?

Reply:
The other family of models using gradient-based optimization.

Reply:
Can we specify this? 

> Non-ensemble models that use gradient-based optimization have a different behavior.